src/ResourceManager/DeviceProvisioningServices/Commands.DeviceProvisioningServices/Cmdlet/IotDps/UpdateAzureRmIotDeviceProvisioningService.cs

```diff
@@ -0,0 +1,224 @@
+            Mandatory = true,
+            ParameterSetName = ResourceCreateUpdateParameterSet,
+            HelpMessage = "Name of the Resource Group")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = ResourceUpdateParameterSet,
+            HelpMessage = "Name of the IoT Device Provisioning Service")]
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = ResourceCreateUpdateParameterSet,
+            HelpMessage = "Name of the IoT Device Provisioning Service")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = InputObjectUpdateParameterSet,
+            HelpMessage = "IoT Device Provisioning Service Tag collection")]
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = ResourceIdUpdateParameterSet,
+            HelpMessage = "IoT Device Provisioning Service Tag collection")]
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            ParameterSetName = ResourceUpdateParameterSet,
+            HelpMessage = "IoT Device Provisioning Service Tag collection")]
+        [ValidateNotNullOrEmpty]
+        public Hashtable Tag { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = InputObjectCreateUpdateParameterSet,
+            HelpMessage = "IoT Device Provisioning Service Allocation policy")]
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = ResourceIdCreateUpdateParameterSet,
+            HelpMessage = "IoT Device Provisioning Service Allocation policy")]
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            ParameterSetName = ResourceCreateUpdateParameterSet,
+            HelpMessage = "IoT Device Provisioning Service Allocation policy")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(new string[] { "Hashed", "GeoLatency", "Static" }, IgnoreCase = true)]
+        public string AllocationPolicy { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = InputObjectUpdateParameterSet,
+            HelpMessage = "Reset IoT Device Provisioning Service Tags")]
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = ResourceIdUpdateParameterSet,
+            HelpMessage = "Reset IoT Device Provisioning Service Tags")]
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = ResourceUpdateParameterSet,
+            HelpMessage = "Reset IoT Device Provisioning Service Tags")]
+        [ValidateNotNullOrEmpty]
```

Comment:
@anusapan nit: this `ValidateNotNullOrEmpty` attribute isn't necessary

Reply:
Done